pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -305,12 +283,35 @@ func parseList(legacyList []runtime.Object) (metainternalversion.ListOptions, ma
 	return options, indexMap, nil
 }
 
-func enrichObject(accessorO, accessorC metav1.Object) {
-	accessorC.SetLabels(accessorO.GetLabels())
+func enrichReturnedObject(originalObj, returnedObj runtime.Object, created bool) (runtime.Object, error) {
+	accessorReturned, err := meta.Accessor(returnedObj)
+	if err != nil {
+		return nil, err
+	}
 
-	ac := accessorC.GetAnnotations()
-	for k, v := range accessorO.GetAnnotations() {
+	accessorOriginal, err := meta.Accessor(originalObj)
+	if err != nil {
+		return nil, err
+	}
+
+	accessorReturned.SetLabels(accessorOriginal.GetLabels())
+
+	ac := accessorReturned.GetAnnotations()
+	if ac == nil {
+		ac = map[string]string{}
+	}
+	for k, v := range accessorOriginal.GetAnnotations() {
 		ac[k] = v
 	}
-	accessorC.SetAnnotations(ac)
+	accessorReturned.SetAnnotations(ac)
+
+	// if the object is created, we need to reset the resource version and UID
+	if created {
+		accessorReturned.SetResourceVersion("")
+		accessorReturned.SetUID("")
+		return returnedObj, nil
+	}
+	accessorReturned.SetResourceVersion(accessorOriginal.GetResourceVersion())
+	accessorReturned.SetUID(accessorOriginal.GetUID())
```

Comment:
```suggestion
	// otherwise, we propagate the original RV and UID
	} else {
		accessorReturned.SetResourceVersion(accessorOriginal.GetResourceVersion())
		accessorReturned.SetUID(accessorOriginal.GetUID())
	}

```